synchronicity/utils.py

```diff
@@ -0,0 +1,27 @@
+import functools
+import sys
+import traceback
+
+def get_tb_filename(tb):
+    return traceback.extract_tb(tb, 1)[0].filename
+
+
+def get_filtered_tb(tb):
+    while tb and "synchronicity" in get_tb_filename(tb):
+        tb = tb.tb_next
+
+    if tb is not None:
+        tb.tb_next = get_filtered_tb(tb.tb_next)
+
+    return tb
+
+def filter_traceback(f):
+    @functools.wraps(f)
+    def f_wrapped(*args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except Exception as exc:
+            tb = sys.exc_info()[2]
```

Comment:
i think you can also do `exc.__traceback__`: https://stackoverflow.com/questions/11414894/extract-traceback-info-from-an-exception-object

Reply:
(marginally nicer since you don't have to import `sys`)

Reply:
good point, updated